lib/monitoring/aws-apigateway/ApiGatewayMetricFactory.ts

```diff
@@ -14,9 +14,9 @@ const ApiGatewayNamespace = "AWS/ApiGateway";
 
 export interface ApiGatewayMetricFactoryProps {
   /**
-   * API to monitor (cannot use IRestApi, since it does not provide API name)
+   * API to monitor
    */
-  readonly api: RestApiBase;
+  readonly api: IRestApi;
```

Comment:
Please add tests that use the interface in `ApiGatewayMonitoring.test.ts`.

Reply:
Absolutely!